src/coreComponents/common/Logger.cpp

```diff
@@ -23,6 +23,32 @@
 namespace geosx
 {
 
+std::string InputError::InsertExMsg( std::string const & originalMsg, std::string const & msgToInsert )
+{
+  std::string newMsg( originalMsg );
+
+  // we try to insert after the "***** Rank N: ", then wh try after "***** ", otherwise, we insert at the top.
+  size_t insertPos;
+  if( ( insertPos = newMsg.find( "***** Rank " ) ) != std::string::npos )
+  {
+    insertPos += 14;
```

Comment:
What does this 14 relate to?
Is it supposed to contain the rank id as well?

Reply:
Indeed, a `insertPos = newMsg.find(':', insertPos + 11 ) + 2;` will be much better here.

Reply:
Then where do these 11 and 2 come from?
Are your properly managing the case where the rank id is 1, 2, 3, 4... characters long?

Reply:
Yes, it is properly managed with `insertPos = newMsg.find(':', insertPos + 11 ) + 2` (I tested it even with a length of 0).
I'll use `strlen()` to improve readability: 
```c++
  size_t insertPos = 0;
  // for readability purposes, we try to insert the message after the "***** Rank N: " or after "***** " instead of at the top.
  static auto constexpr rankLogStartStr =  "***** Rank ";
  static auto constexpr rankLogEndStr =  ": ";
  static auto constexpr simpleLogStartStr =  "***** ";
  if( ( insertPos = newMsg.find( rankLogStartStr ) ) != std::string::npos )
  {
    insertPos = newMsg.find( rankLogEndStr, insertPos + strlen( rankLogStartStr ) ) + strlen( rankLogEndStr );
  }
  else if( ( insertPos = newMsg.find_last_of( simpleLogStr ) ) != std::string::npos )
  {
    insertPos += strlen( simpleLogStartStr );
  }
  newMsg.insert( insertPos, msgToInsert );
```